Bottom: 89b109b9172f2fd26f7feaca314718e1127b4e64
Top:    6fb26f02c06c27cb2b94fd21d21273260c3c4563
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-17 12:39:37 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index 1d76df9..cdf1ea6 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -391,7 +391,7 @@ int socket(int domain, int type, int protocol)
 	if (ret >= 0) {
 		if (fork_support) {
 			rclose(ret);
-			ret = real_socket(domain, type, protocol);
+			ret = real.socket(domain, type, protocol);
 			if (ret < 0)
 				return ret;
 			fd_store(index, ret, fd_fork);
@@ -863,6 +863,7 @@ pid_t fork(void)
 
 	param = 1;
 	rsetsockopt(dfd, IPPROTO_TCP, TCP_NODELAY, &param, sizeof param);
+	set_rsocket_options(dfd);
 
 	msg = 0;
 	ret = real.write(sfd, &msg, sizeof msg);
